src/regular-expression-comment.ts

```diff
@@ -24,10 +24,28 @@ export default {
           const regexLine = node.loc?.start.line;
           if (regexLine) {
             const previousLine = regexLine - 1;
-
-            const hasRegexComment = comments.find(
-              (comment) => comment.loc?.start.line === previousLine || comment.loc?.start.line === regexLine,
-            );
+            const hasRegexComment = comments.find((comment) => {
+              if (!comment.loc) {
+                return false;
+              }
+              const regExp = /[a-zA-Z]/gu;
+              const commentRegExp = /^(?:\/{2,}|\/\*+)/gu;
+              const hasComment = regExp.test(comment.value.trim());
+              const previousLineComment = context.sourceCode.getLines()[previousLine - 1];
```

Comment:
this looks potentially hazardous performance wise, calling `context.sourceCode.getLines()` for every regular expression.  Since its not going to change, I'd get that value up at the top where you also get the comments.